server/geoip_service.py

```diff
@@ -69,23 +70,45 @@ def __init__(self):
             Download the geoip database to a file. If the downloaded file is not
         a valid gzip file, then it does NOT overwrite the old file.
         """
+        if not config.GEO_IP_LICENSE_KEY:
+            self._logger.warning(
+                "GEO_IP_LICENSE_KEY not set! Unable to download GeoIP database!"
+            )
+            return
 
         self._logger.info("Downloading new geoip database")
 
         # Download new file to a temp location
-        temp_file_path = "/tmp/geoip.mmdb.gz"
-        await self._download_file(config.GEO_IP_DATABASE_URL, temp_file_path)
+        temp_file_path = "/tmp/geoip.mmdb.tar.gz"
```

Comment:
maybe use https://docs.python.org/3/library/tempfile.html ?

Reply:
I don't see a reason to. The purpose is just so it doesn't overwrite the existing file while downloading. There's no need to use a module just to add `/tmp` to the beginning of a file path :P

Reply:
If something goes wrong, file could stay locked for a while. I just propose the tempfile.

Reply:
How would it get locked? It's opened in a context manager and this function is supposed to be called once every 3 weeks...

Reply:
Ok. NP.